### PR TITLE
Dicom update cleanup function

### PR DESCRIPTION
**Pull request by arunmk-ms**
## Description
A new cleanup function to remove the newly added blob in case of a failure.

## Related issues
Addresses [[issue #102010](https://microsofthealth.visualstudio.com/Health/_workitems/edit/102010)].

## Testing
Manual testing

### CODE REVIEW

File: src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Orchestration.cs [hunk 1]
```diff
@@ -131,6 +132,24 @@ public partial class UpdateDurableFunction
             {
                 logger.LogInformation("Update operation completed successfully");
             }
+
+            if (input.TotalNumberOfInstanceUpdated > 0)
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Orchestration.cs:136]: Don't we need use replaySafecounter?

Comment (arunmk-ms) [reply src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Orchestration.cs:136]: Good catch. I missed it while merging from main. Added it back.

File: src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Activity.cs [hunk 2]
```diff
@@ -187,5 +187,42 @@ public async Task DeleteOldVersionBlobAsync([ActivityTrigger] IDurableActivityCo
...
+            {
+                CancellationToken = default,
+                MaxDegreeOfParallelism = _options.MaxParallelThreads,
+            },
+            async (fileIdentifier, token) =>
+            {
+                await _updateInstanceService.DeleteInstanceBlobAsync(fileIdentifier.NewVersion.Value, token);
+            });
+
+        logger.LogInformation("New blobs deleted successfully. Total size {TotalCount}", fileIdentifiers.Count);
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Activity.cs:223]: Could you update to fileIdentifiers.Where(f => f.NewVersion.HasValue) count. Can you do the same for the previous activity

File: src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Activity.cs [hunk 3]
```diff
@@ -184,8 +185,46 @@ public async Task DeleteOldVersionBlobAsync([ActivityTrigger] IDurableActivityCo
...
+            },
+            async (fileIdentifier, token) =>
+            {
+                await _updateInstanceService.DeleteInstanceBlobAsync(fileIdentifier.NewVersion.Value, token);
+            });
+
+        logger.LogInformation("New blobs deleted successfully. Total size {TotalCount}", fileCount);
+    }
+
+
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Activity.cs:228]: nit: extra line

Comment (arunmk-ms) [reply src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Activity.cs:228]: Removed

**Review by bcarthic — APPROVED** [review]

**Pull request merged**